src/python/pants/engine/target.py

```diff
@@ -584,6 +584,37 @@ def types(self) -> Tuple[Type[Target], ...]:
         return tuple(self.aliases_to_types.values())
 
 
+def generate_subtarget(base_target: Target, *, full_file_name: str) -> Target:
```

Comment:
We may want to make this a rule in the future. For now, I wanted to keep it simple—easier to test and no wrapper request type needed.

Reply:
I'm pretty sure calling a method in python avoids doing a back and forth crossing of the FFI boundary as opposed to an `await Get(...)`, which seems prone to becoming slower if compounded over a deep call stack of these crossings. That's in addition to the other benefits you described.

As long as these methods are type-checked, it's still probably just as easy to do utterly ridiculous things like compile them to WebAssembly later on down the line (I believe -- haven't thought that through).